plugins/uefi-dbx/fu-uefi-dbx-device.c

```diff
@@ -128,10 +128,19 @@ fu_uefi_dbx_device_ensure_checksum(FuUefiDbxDevice *self, GError **error)
 
 	/* add the last checksum to the device */
 	sigs = fu_firmware_get_images(dbx);
-	if (sigs->len > 0) {
-		FuEfiSignature *sig = g_ptr_array_index(sigs, sigs->len - 1);
+
+	for (guint i = sigs->len - 1; i > 0; i--) {
```

Comment:
what happens if `sigs->len == 0`?

Reply:
> what happens if sigs->len == 0?

Underflows, wrapping around and looping forever.

Updated.